src/listeners.ts

```diff
@@ -0,0 +1,582 @@
+    const key_is_z = (
+        keydown_event.key === "z" ||
+        keydown_event.key === "Z" ||
+        keydown_event.code === "KeyZ"
+    );
+
+    if (ctrl && key_is_z) {
+        keydown_event.preventDefault();
+        if (shift) {
+            ulabel.redo();
+        } else {
+            ulabel.undo();
+        }
+        return false;
+    } else {
+        const current_subtask = ulabel.get_current_subtask();
+        switch (keydown_event.key) {
+            case "Escape":
+                // If in erase or brush mode, cancel the brush
+                if (current_subtask.state.is_in_erase_mode) {
+                    ulabel.toggle_erase_mode();
+                } else if (current_subtask.state.is_in_brush_mode) {
+                    ulabel.toggle_brush_mode();
+                } else if (current_subtask.state.starting_complex_polygon) {
+                    // If starting a complex polygon, undo
+                    ulabel.undo();
+                } else if (current_subtask.state.is_in_progress) {
+                    // If in the middle of drawing an annotation, cancel the annotation
+                    ulabel.cancel_annotation();
+                }
+                break;
+        }
+    }
+}
+
+/**
+ * Create listeners for a ULabel instance.
+ * Consider breaking out long handlers.
+ *
+ * @param ulabel ULabel instance
+ */
+export function create_ulabel_listeners(
+    ulabel: ULabel,
+) {
+    // ================= Mouse Events in the ID Dialog =================
+    const id_dialog = $(".id_dialog");
+    id_dialog.on(
+        "mousemove" + ULABEL_NAMESPACE,
+        function (mouse_event) {
+            if (!ulabel.get_current_subtask()["state"]["idd_thumbnail"]) {
+                ulabel.handle_id_dialog_hover(mouse_event);
+            }
+        },
+    );
+
+    // ================= Mouse Events in the Annotation Container =================
+    const annbox = $("#" + ulabel.config["annbox_id"]);
+
+    // Detect and record mousedown
+    annbox.on(
+        "mousedown" + ULABEL_NAMESPACE,
+        function (click_event) {
+            ulabel.handle_mouse_down(click_event);
+        },
+    );
+
+    // Prevent default for auxclick
+    $(document).on(
+        "auxclick" + ULABEL_NAMESPACE,
+        ulabel.handle_aux_click,
+    );
+
+    // Detect and record mouseup
+    $(document).on(
+        "mouseup" + ULABEL_NAMESPACE,
+        ulabel.handle_mouse_up.bind(ulabel),
+    );
+
+    $(window).on(
+        "click" + ULABEL_NAMESPACE,
+        function (click_event) {
+            if (click_event.shiftKey) {
+                click_event.preventDefault();
+            }
+        },
+    );
+
+    // Mouse movement has meaning in certain cases
+    annbox.on(
+        "mousemove" + ULABEL_NAMESPACE,
+        function (move_event) {
+            ulabel.handle_mouse_move(move_event);
+        },
+    );
+
+    // ================= Uncategorized =================
```

Comment:
Top sections already had the bar, but there weren't any bars for the remainder, so I threw this in. 

If you can think of logical groupings, we could add more bars. Alternatively, we could just make them be functions. :shrug: